api/base/serializers.py

```diff
@@ -88,6 +89,26 @@ def to_internal_value(self, data):
         return super(JSONAPIListField, self).to_internal_value(data)
 
 
+class AuthorizedCharField(ser.CharField):
```

Comment:
I'm not sure this is necessary for now. (1) This is only used in one place. I think a `SerializerMethodField` will achieve what we want there. (2) This is not composable. What happens when we need `AuthorizedIntegerField`, `AuthorizedBooleanField`, etc.?